### DISCUSSION TITLE
Versionamento Api

### DISCUSSION THREAD

**Discussion by jucelinopinheiro** [balta-io/2811#27] [Q&A] [answered]
Na aula de versionamento é mostrada de duas forma de nomear a versão da API de uma forma simples e clara, quando utilizamos o versionamento diretamente no Método tranquilo no exemplo [HttpGet(“v1/categories”)] no caso de uma versão 2 [HttpGet(“v2/categories”)]. Minha dúvida surgiu quando pré-fixamos na Rota a versão [Route(“v1”)]. Como fica o código abaixo se tivermos uma [Route(“v2”)].

    **[ApiController]**
    **[Route("v1")]** //pre fixando a rota do controller
     public class HomeController:ControllerBase
    {
        [HttpGet("categories")]//agora localhost:porta/v1/categories
        public IActionResult Get(
			[FromServices]BlogDataContext context)
       {
        return Ok();
       }
     }

**Comment by andrebaltieri** [comment, marked as answer]
@jucelinopinheiro você pode ter múltiplas rotas para o mesmo endpoint, ou se mudar muita coisa, pode ter dois controllers, um atendendo V1 e outro atendendo V2.
